modules/library/render/src/main/java/org/geotools/renderer/composite/BlendComposite.java

```diff
@@ -0,0 +1,653 @@
+                // Da' = Sa + Da - Sa.Da
+                // @formatter:on
+
+                result[RED] = (sr + dr - ((2 * Math.min(sr * da, dr * sa) + UBYTE_MAX_VALUE) >> SHIFT8));
+                result[GREEN] = (sg + dg - ((2 * Math.min(sg * da, dg * sa) + UBYTE_MAX_VALUE) >> SHIFT8));
+                result[BLUE] = (sb + db - ((2 * Math.min(sb * da, db * sa) + UBYTE_MAX_VALUE) >> SHIFT8));
+                result[ALPHA] = (sa + da - ((sa * da + UBYTE_MAX_VALUE) >> SHIFT8));
+            }
+
+        },
+        Exclusion("exclusion") {
+
+            @Override
+            public void perform(int sr, int sg, int sb, int sa, int dr, int dg, int db, int da,
+                    int[] result) {
+                // @formatter:off
+                // Dca' = (Sca.Da + Dca.Sa - 2.Sca.Dca) + Sca.(1 - Da) + Dca.(1 - Sa)
+                // Da' = Sa + Da - Sa.Da
+                // @formatter:on
+
+                int s1a = UBYTE_MAX_VALUE - sa;
+                int d1a = UBYTE_MAX_VALUE - da;
+
+                result[RED] = ((sr * da + dr * sa - 2 * sr * dr + sr * d1a + dr * s1a + UBYTE_MAX_VALUE) >> SHIFT8);
+                result[GREEN] = ((sg * da + dg * sa - 2 * sg * dg + sg * d1a + dg * s1a + UBYTE_MAX_VALUE) >> SHIFT8);
+                result[BLUE] = ((sb * da + db * sa - 2 * sb * db + sb * d1a + db * s1a + UBYTE_MAX_VALUE) >> SHIFT8);
+                result[ALPHA] = (sa + da - ((sa * da + UBYTE_MAX_VALUE) >> SHIFT8));
+            }
+
+        };
+
+        String name;
+
+        BlendingMode(String name) {
+            this.name = name;
+        }
+
+        /**
+         * Performs the color blending on the given pixels, assuming the source colors are
+         * pre-multiplied
+         * 
+         * @param sr
+         * @param sg
+         * @param sb
+         * @param sa
+         * @param dr
+         * @param dg
+         * @param db
+         * @param da
+         * @param result
+         */
+        public abstract void perform(int sr, int sg, int sb, int sa, int dr, int dg, int db,
+                int da, int[] result);
+
+        public String getName() {
+            return name;
+        }
+
+        /**
+         * Looks up a blending mode by its SVG standard name (as opposed to the enum name, which for
+         * example cannot contain hyphens)
+         * 
+         * @param name The standard name
+         * @return The corresponding blending mode, or null if not found
+         */
+        public static BlendingMode lookupByName(String name) {
+            for (BlendingMode mode : values()) {
+                if (mode.getName().equals(name)) {
+                    return mode;
+                }
+            }
+
+            return null;
+        }
+
+        /**
+         * Returns the list of blending modes standard names
+         * 
+         * @return The list of blending mode names
+         */
+        public static List<String> getStandardNames() {
```

Comment:
I want something with predictable order. The standard java way to return all the enum values gives you an array, not a Set... so not even the language builders agree with this Set idea.